presto-main/src/main/java/io/prestosql/dispatcher/DispatchManager.java

```diff
@@ -271,9 +271,9 @@ public BasicQueryInfo getQueryInfo(QueryId queryId)
         return queryTracker.getQuery(queryId).getBasicQueryInfo();
     }
 
-    public QueryInfo getFullQueryInfo(QueryId queryId)
+    public Optional<QueryInfo> getFullQueryInfo(QueryId queryId)
```

Comment:
This isn't consistent with other methods here: `getQuery(QueryId)`, `getQueryInfo(QueryId)`.

IMO renaming this method to `tryGetFullQueryInfo` would be better... Unless I fail to convince you.

Reply:
I would prefer to remove `getQuery`, rename `tryGetQuery` to `getQuery` and remove usage of `NoSuchElementException`. `Optional` is much better than handling an exception here.